flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java

```diff
@@ -1275,7 +1314,31 @@ public static long timestampCeil(TimeUnitRange range, long ts, TimeZone tz) {
                 int days = (int) (utcTs / MILLIS_PER_DAY + EPOCH_JULIAN);
                 return julianDateFloor(range, days, false) * MILLIS_PER_DAY - offset;
             default:
-                // for MINUTE and SECONDS etc...,
+                // it is more effective to use arithmetic Method
+                throw new AssertionError(range);
+        }
+    }
+
+    public static int timestampCeilForHighPrecision(TimeUnitRange range, int ns) {
+        switch (range) {
+            case NANOSECOND:
+                return ns;
+            case MICROSECOND:
+                return ceilForHighPrecision(ns);
+            case MILLISECOND:
```

Comment:
here and for `timestampFloorForHighPrecision` we have only one line different
can we make logic more generic here?